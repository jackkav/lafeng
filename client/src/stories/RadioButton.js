import React from 'react'
import { ReactRadioButtonsGroup, ReactRadioButton } from 'react-radio-buttons-group'

const RadioButton = ({onChange, options}) => (
  <ReactRadioButtonsGroup group="sex" onChange={onChange}>
    {options.map(option => (<ReactRadioButton value={option.value} key={option.value}>{option.label}</ReactRadioButton>))}
  </ReactRadioButtonsGroup>
)
RadioButton.defaultProps = {
  onChange: value => console.log(value),
  options: [{label: 'Male ðŸ‘¨', value: 'male'}, {label: 'Female ðŸ‘©', value: 'female'}]
}
RadioButton.propTypes = {
  onChange: React.PropTypes.func,
  options: React.PropTypes.array
}

export default RadioButton
