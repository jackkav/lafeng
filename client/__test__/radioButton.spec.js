import React from 'react'
import RadioButton from '../src/components/RadioButton'
import renderer from 'react-test-renderer'

it('renders correctly without any props', () => {
  const tree = renderer.create(
    <RadioButton />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly with two options', () => {
  const input = [{label: 'a', value: 'a'}, {label: 'b', value: 'b'}]
  const tree = renderer.create(
    <RadioButton options={input} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
