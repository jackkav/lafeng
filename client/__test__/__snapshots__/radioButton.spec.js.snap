exports[`test renders correctly with two options 1`] = `
<div
  className="react-radio-buttons-group">
  <div
    className="react-radio-buttons-group__item">
    <div
      className="react-radio-button">
      <input
        className="react-radio-button__input"
        id="sex[a]"
        name="sex"
        onChange={[Function]}
        type="radio"
        value="a" />
      <label
        className="react-radio-button__control"
        htmlFor="sex[a]">
        a
      </label>
    </div>
  </div>
  <div
    className="react-radio-buttons-group__item">
    <div
      className="react-radio-button">
      <input
        className="react-radio-button__input"
        id="sex[b]"
        name="sex"
        onChange={[Function]}
        type="radio"
        value="b" />
      <label
        className="react-radio-button__control"
        htmlFor="sex[b]">
        b
      </label>
    </div>
  </div>
</div>
`;

exports[`test renders correctly without any props 1`] = `
<div
  className="react-radio-buttons-group">
  <div
    className="react-radio-buttons-group__item">
    <div
      className="react-radio-button">
      <input
        className="react-radio-button__input"
        id="sex[male]"
        name="sex"
        onChange={[Function]}
        type="radio"
        value="male" />
      <label
        className="react-radio-button__control"
        htmlFor="sex[male]">
        Male ðŸ‘¨
      </label>
    </div>
  </div>
  <div
    className="react-radio-buttons-group__item">
    <div
      className="react-radio-button">
      <input
        className="react-radio-button__input"
        id="sex[female]"
        name="sex"
        onChange={[Function]}
        type="radio"
        value="female" />
      <label
        className="react-radio-button__control"
        htmlFor="sex[female]">
        Female ðŸ‘©
      </label>
    </div>
  </div>
</div>
`;
